daemon/config/config_unix.go

```diff
@@ -96,12 +96,29 @@ func verifyDefaultCgroupNsMode(mode string) error {
 	return nil
 }
 
+func VerifyUlimits(ulimits map[string]*units.Ulimit) error {
+	for _, v := range ulimits {
+		if _, err := v.GetRlimit(); err != nil {
+			return err
+		}
+		if v.Soft > v.Hard {
```

Comment:
Looks like the validation is slightly different than the one used to parse the flag; https://github.com/docker/go-units/blob/519db1ee28dcc9fd2474ae59fca29a810482bfb1/ulimit.go#L99-L106

Ideally, we should make sure that both follow the same rules; perhaps the code for this validation should be in the go-units package? wdyt?

Reply:
I think so! `go-units` should provide validation methods along with parsing ones.